watcher/local/watcher.go

```diff
@@ -354,14 +364,61 @@ func (ch *ch) handleEventsFromChain(registerer channel.Registerer, progresser ch
 	}
 }
 
+func (ch *ch) handleRegisteredEvent(
+	ctx context.Context,
+	e *channel.RegisteredEvent,
+	registerer channel.Registerer,
+	chRegistry *registry,
+) {
+	parent := ch
+	if ch.isSubChannel() {
+		parent = ch.parent
+	}
+
+	// This lock ensures, when there are one or more sub-channels and
+	// adjudicator event is received for each channel, the events are
+	// processed one after the other.
+	//
+	// Even if older states have been registered for more than one
+	// channel in the same family, the channel tree will be registered
+	// only once.
+	parent.subChsAccess.Lock()
```

Comment:
does this lock recursively for subchannels? I think it's strange to sometimes lock this channel's sub-channels, and sometimes only lock this channel's parent's sub-channels. I can't validate this without some additional effort.